src/entrypoints/entrypoints.ts

```diff
@@ -34,19 +34,24 @@ export class NetworkEntrypoint {
     constructor(options: {
         networkProviderUrl: string;
         networkProviderKind: string;
+        networkProviderOptions?: NetworkProviderConfig;
         chainId: string;
         clientName?: string;
         withGasLimitEstimator?: boolean;
         gasLimitMultiplier?: number;
     }) {
+        options.networkProviderOptions = options.networkProviderOptions || {};
+
+        if (options.clientName) {
+            options.networkProviderOptions.clientName = options.clientName;
```

Comment:
Such a line:

```
options.networkProviderOptions.[...] = ...
```

Will mutate the input argument `networkProviderOptions`, which might cause downstream issues (if that input argument is re-used by the caller). Maybe clone the options (below is just an example, not an actual suggestion)?

```
networkProviderOptions = { ...options.networkProviderOptions, ...{ clientName: options.clientName } };
```

Maybe a cleaner way would be to simply drop the input argument `clientName`, and only have `networkProviderConfig`? However, that would be a breaking change, and we shouldn't apply it (yet!).

Reply:
About the `clientName` and `User-Agent` header, my opinion is:

- `clientName` is good for general usage, as you want to collect stats. It's easy to set and abstracts the technical implications. And you already build a custom User-Agent from that, that looks like a good approach.
- More advanced users may want to force a very specific `User-Agent`. This shouldn't be altered later in the Network Provider.
- One of them must be specified for sure, otherwise you get the console error message and the "unknown" fallback. With a priority to be discussed (probably User-Agent > clientName).